apps/nextjs-website/src/lib/api.ts

```diff
@@ -39,3 +46,68 @@ export const getHeaderProps = async (): Promise<HeaderProps> => {
   const navigation = await getNavigation('main-navigation', appEnv);
   return makeHeaderProps(navigation, header);
 };
+
+const generateLinks = (
+  section: FooterSection,
+  isSocialLink = false
+): ReadonlyArray<FooterLinksType> => {
+  const filteredLinks = section.links.filter((link) =>
+    isSocialLink
+      ? link.linkType === 'social' && link.icon
+      : ['internal', 'external'].includes(link.linkType)
+  );
+
+  return filteredLinks.map((link) => ({
+    label: link.text ?? link.href,
+    href: link.href,
+    ariaLabel: link.ariaLabel ?? link.href,
+    linkType: link.linkType as LinkType,
+    ...(isSocialLink && { icon: link.icon as keyof typeof MuiIcons }),
+  }));
+};
+
+export const getFooterProps = async (): Promise<
+  Omit<FooterProps, 'onLanguageChanged'>
+> => {
+  // Omitting onLanguageChanged since we cannot pass event handler functions to server component (in this case Layout)
+  const footerAPIRes = await getFooter(appEnv);
+  const footerData = footerAPIRes.data.attributes;
+
+  return {
+    companyLink: {
+      ariaLabel:
+        footerData.companyLink.ariaLabel ?? footerData.companyLink.href,
+      href: footerData.companyLink.href,
+    },
+    legalInfo: footerData.legalInfo, // TODO: This is rich text (basically markdown) so we should parse it into HTML --> RFC needed to determine how we're going to do this
+    links: {
+      aboutUs: {
+        title: footerData.links_aboutUs.title ?? '',
```

Comment:
If the title is required, you can ensure its presence [during the `decode` phase](https://github.com/pagopa/b2b-portals/pull/80/files#diff-f6b7d09abae598bca9c0b7f72462792c9fc8210c63de7393327a4ad409d5d614R21) instead of relying on a default value (`''`)

Reply:
The "About us" section currently wants no title as seen in the [legacy site](https://notifichedigitali.pagopa.it/).

The component requires a value though, so we just opted to turn an unaccepted `null` value into an empty string.